docs/source/templates/class_api.rst

```diff
@@ -0,0 +1,197 @@
+..
+  Use the following convention for headings:
+
+    # with overline, for parts (collections of chapters)
+
+    * with overline, for chapters
+
+    = for sections
+
+    - for subsections
+
+    ^ for subsubsections
+
+    " for paragraphs
+
+..
+  Class / method / container name)
+  for free functions that are callable, preserve the naming convention, `view_alloc()`
+
+``CoolerView``
+=====================
```

Comment:
the length of === should be fixed to match the word otherwise this is misleading since sphinx would raise an exception for this